ARCHITECTURE.md

```diff
@@ -336,9 +337,8 @@ without breaking anything.
 
 It might seem weird that there are `*StringNoLocale` and `*StringWithLocale`
 functions, rather than just `*String` functions with an optional `locale`
-parameter. This was a lengthy discussion where nobody left happy, but the
-reasons had to do with forcing people to make a conscious choice not to set a
-locale.
+parameter. This was a lengthy internal discussion where the desicion was to nudge 
```

Comment:
```suggestion
parameter. This was a lengthy internal discussion where the decision was to nudge 
```

Reply:
The previous sentence with "It might seem weird ...."  
Maybe just go for the facts instead of opinions and go

"The decision for having `*StringNoLocale` and `*StringWithLocale` functions instead of `*String` functions with an optional `locale` was based on an length internal discussion where ..."